.github/workflows/integration.yml

```diff
@@ -0,0 +1,102 @@
+# Run secret-dependent integration tests only after /ok-to-test approval
+on:
+  pull_request:
+  repository_dispatch:
+    types: [ok-to-test-command]
+
+name: Integration tests
+
+jobs:
+  # Branch-based pull request
+  integration-trusted:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository
```

Comment:
I wonder if it is worth adding in a comment here explaining what this does and why it is important for "trusted builds"